telemetry/specs/telemetry_report.mdk

```diff
@@ -424,47 +439,107 @@ The Destination Port is user configurable. The expectation is that the same
 Destination Port value will be used for all telemetry reports in a
 particular deployment.
 
-## Telemetry Report Header (16+ octets)
+## Telemetry Report Header (Ver 2.0) (20+ octets)
+
 
-`
        0                   1                   2                   3
        0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
       +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-      |  Ver  |Length |NProt| RepMdBits | Reserved  |D|Q|F|   hw_id   |
-      +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-      |                           Switch id                           |
-      +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-      |                        Sequence Number                        |
-      +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-      |                       Ingress Timestamp                       |
+      |  Ver  |   hw_id   |           Sequence Number                 |
       +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-      |                  Variable Optional Metadata                   |
-      +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-`
+      |                            Node id                            |
+      +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+<-+
+      |RepType| InType| Report Length |   MD Length   |D|Q|F|I|  Rsvd |  |
+      +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+  R
+      |            RepMdBits          |      Domain Specific ID       |  E
+      +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+  P
+      |            DSMdBits           |          DSMdstatus           |  O                        
+      +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+  R
+      |          Variable Optional Baseline & DS Metadata             |  T                   
```

Comment:
This seems to be overly constrained by not allowing for Truncated Packet or  Additional DS Extension Data in the same report.

Allowing for both within the same report will enable the full spectrum of data.  Plus it will allow an application that only knows about the packet fragment to read that and skip the DS extension data.

This will require that the "IntType" and Report Length would need to be duplicated for Domain Specific Extension Data and the packet fragment.